pkg/cmd/roachtest/test_runner.go

```diff
@@ -1191,7 +1194,12 @@ func (r *testRunner) runTest(
 		if t.Failed() {
 			s = "with failure(s)"
 		}
-		t.L().Printf("test completed %s", s)
+		status := fmt.Sprintf("test completed %s", s)
+		// Attempt to annotate the test completion on Grafana.
+		if err := c.AddGrafanaAnnotation(ctx, t.L(), grafana.AddAnnotationRequest{Text: status}); err != nil {
```

Comment:
Should we include the test name in this as well? I think the status above only tells us if it failed, but doesn't include the name of the test like `grafanaAnnotateTestStart`.

Also curious,  I see annotations support time ranges (https://grafana.com/docs/grafana/latest/developers/http_api/annotations/). Would that make sense for something like a test? It also then only requires one call that specifies.

Only negative I can think of is that the time range annotations possibly make reading the graphs a little harder.

Reply:
I noted the API you wrote does support time ranges. Any reason for not using those here?

Reply:
>Should we include the test name in this as well?

Sure! I don't see the harm in doing that, done.

> I noted the API you wrote does support time ranges

Unfortunately, Grafana seems to give precedence to the last added annotation when you're trying to hover over two overlapping annotations. So making this a time range would make all other annotations added unselectable. 

I already notice this with some of the mixed version roachtests that run a workload through the entire test, and from what I can tell the only way to view the annotations under it is to delete the workload annotation.

Reply:
[Example](https://grafana.testeng.crdb.io/d/crdb-console-sql-byname/crdb-console-sql-by-test-run-and-name?from=1713782266663&to=1713784066663&var-DS_PROMETHEUS=Prometheus&var-test_run_id=teamcity-14927289&var-test_name=cdc-mixed-versions&var-cluster=teamcity-14927289-1713764899-47-n5cpu4&orgId=1) of what I mean. You can't view any of the annotations besides the kafka workload running through the entire test.

Reply:
Ah thanks for explaining, I figured something like that may be the case, in that regard I opt for the single ones, rather than the range.